### PR TITLE
Rotd fix take2

### PR DESCRIPTION
**Pull request by jpa198**

### CODE REVIEW

**Comment by jpa198** [comment]
I thought I needed the python rspectra fix in the rotd PR, but I don't so I've made this one instead
Verification below:

<p>I've tested this with a copy of Hossack I had lying around.<br>
The new figures line up with those of the adjusted old figures, and are a lot closer to geom than the original calculations.</p>
<p>Old:</p>

station | component | pSA_0.02 | pSA_0.05 | pSA_0.1 | pSA_0.2 | pSA_0.3 | pSA_0.4 | pSA_0.5 | pSA_0.75 | pSA_1.0 | pSA_2.0 | pSA_3.0 | pSA_4.0 | pSA_5.0 | pSA_7.5 | pSA_10.0
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
WRAV | geom | 0.06013336675273 | 0.089115745322454 | 0.125133181113855 | 0.121005303566589 | 0.144512778030697 | 0.103801783325256 | 0.080523663769336 | 0.067167579184118 | 0.028598548143978 | 0.01028351531141 | 0.004587755557469 | 0.001993342513191 | 0.001102070004063 | 0.000443169802414 | 0.000250627974779
WRAV | rotd50 | 6.2335278555554E-07 | 5.72149302246633E-06 | 2.96559252638543E-05 | 0.000149028614283 | 0.000344619064028 | 0.00042019840063 | 0.000535445191359 | 0.000950297921658 | 0.000755494339119 | 0.001070922696095 | 0.001041079503484 | 0.000849442741323 | 0.000697894240638 | 0.000575025983932 | 0.000597973586903

<p>Old with w^2 applied:</p>

station | component | pSA_0.02 | pSA_0.05 | pSA_0.1 | pSA_0.2 | pSA_0.3 | pSA_0.4 | pSA_0.5 | pSA_0.75 | pSA_1.0 | pSA_2.0 | pSA_3.0 | pSA_4.0 | pSA_5.0 | pSA_7.5 | pSA_10.0
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
WRAV | geom | 0.06013336675273 | 0.089115745322454 | 0.125133181113855 | 0.121005303566589 | 0.144512778030697 | 0.103801783325256 | 0.080523663769336 | 0.067167579184118 | 0.028598548143978 | 0.01028351531141 | 0.004587755557469 | 0.001993342513191 | 0.001102070004063 | 0.000443169802414 | 0.000250627974779
WRAV | rotd50 | 0.061522453957503 | 0.090350196344537 | 0.117076900201006 | 0.147085346741202 | 0.151166836934552 | 0.103679799604779 | 0.084554115474825 | 0.066695570132952 | 0.02982572101747 | 0.010569583354605 | 0.004566685710875 | 0.002095915954559 | 0.001102070411023 | 0.000403575394258 | 0.000236070509801

<p>New:</p>

station | component | pSA_0.02 | pSA_0.05 | pSA_0.1 | pSA_0.2 | pSA_0.3 | pSA_0.4 | pSA_0.5 | pSA_0.75 | pSA_1.0 | pSA_2.0 | pSA_3.0 | pSA_4.0 | pSA_5.0 | pSA_7.5 | pSA_10.0
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
WRAV | geom | 0.06013336675273 | 0.089115745322454 | 0.125133181113855 | 0.121005303566589 | 0.144512778030697 | 0.103801783325256 | 0.080523663769336 | 0.067167579184118 | 0.028598548143978 | 0.01028351531141 | 0.004587755557469 | 0.001993342513191 | 0.001102070004063 | 0.000443169802414 | 0.000250627974779
WRAV | rotd50 | 0.061522453957503 | 0.090350196344537 | 0.117076900201006 | 0.147085346741202 | 0.151166836934552 | 0.103679799604779 | 0.084554115474825 | 0.066695570132952 | 0.02982572101747 | 0.010569583354605 | 0.004566685710875 | 0.002095915954559 | 0.001102070411023 | 0.000403575394258 | 0.000236070509801

**Review by claudio525 — APPROVED** [review]

File: IM_calculation/IM/im_calculation.py [hunk 1]
```diff
@@ -84,19 +84,19 @@ def check_rotd(comps_to_store: Iterable[Components]) -> bool:
...
-    :param spectral_displacements: An array with shape [periods.size, nt, 2] where nt is the number of timesteps in the original waveform
+    Calculates rotd for given accelerations
+    :param accelerations: An array with shape [periods.size, nt, 2] where nt is the number of timesteps in the original waveform
     :param comps_to_store: A list of components to store
     :param func: The function to apply to the rotated waveforms. Defaults to taking the maximum absolute value across all rotations (used by PGA, PGV, pSA)
     :return: A dictionary with the comps_to_store as keys, and 1d arrays of shape [periods.size] containing the rotd values
     """
     # Selects the first two basic components. get_comps_to_calc_and_store makes sure that the first two are 000 and 090
-    rotd = intensity_measures.get_rotations(spectral_displacements[:, :, [0, 1]], func)
+    rotd = intensity_measures.get_rotations(accelerations[:, :, [0, 1]], func)
```
Comment (jpa198) [inline IM_calculation/IM/im_calculation.py:99]: While we're here, should this be changed to [..., [0,1]]?
Which should mean we don't need to worry about the number of axis'

Comment (jasonmotha) [reply IM_calculation/IM/im_calculation.py:99]: That actually seems reasonable.

File: IM_calculation/IM/im_calculation.py [hunk 2]
```diff
@@ -84,19 +84,19 @@ def check_rotd(comps_to_store: Iterable[Components]) -> bool:
...
-    spectral_displacements,
+    accelerations,
     comps_to_store: List[Components],
     func=lambda x: np.max(np.abs(x), axis=1),
 ):
     """
-    Calculates rotd for given spectral displacements
-    :param spectral_displacements: An array with shape [periods.size, nt, 2] where nt is the number of timesteps in the original waveform
+    Calculates rotd for given accelerations
+    :param accelerations: An array with shape [periods.size, nt, 2] where nt is the number of timesteps in the original waveform
```
Comment (jasonmotha) [inline IM_calculation/IM/im_calculation.py:93]: Can you update this comment as per my previous comments?

**Review by jasonmotha — APPROVED** [review]
Approval conditional on wiki page.

**Comment by jpa198** [comment]
Wiki page with documentation. Further input welcomed.
https://wiki.canterbury.ac.nz/display/QuakeCore/pSA+changes

File: IM_calculation/IM/im_calculation.py [hunk 3]
```diff
@@ -226,7 +226,7 @@ def calc_CAV(waveform, times, im, comps_to_store, comps_to_calculate):
         func = lambda x: intensity_measures.get_cumulative_abs_velocity_nd(
             np.squeeze(x), times=times
         )
-        rotd = calculate_rotd(np.expand_dims(waveform, 0), comps_to_store, func)
+        rotd = calculate_rotd(waveform, comps_to_store, func)
```
Comment (sungeunbae) [inline IM_calculation/IM/im_calculation.py:229]: just being a bit pedantic here. `func=func` will be nice to be consistent with other usages.

File: IM_calculation/IM/im_calculation.py [hunk 4]
```diff
@@ -236,7 +236,7 @@ def calc_MMI(waveform, im, comps_to_store, comps_to_calculate):
     values = array_to_dict(value, comps_to_calculate, im, comps_to_store)
     if check_rotd(comps_to_store):
         func = lambda x: intensity_measures.calculate_MMI_nd(np.squeeze(x))
-        rotd = calculate_rotd(np.expand_dims(waveform, 0), comps_to_store, func)
+        rotd = calculate_rotd(waveform, comps_to_store, func)
```
Comment (sungeunbae) [inline IM_calculation/IM/im_calculation.py:239]: `func=func` as above

**Review by sungeunbae — APPROVED** [review]
Very insignificant comments - Happy to accept as-is.

**Comment by jasonmotha** [comment]
> Wiki page with documentation. Further input welcomed.
> https://wiki.canterbury.ac.nz/display/QuakeCore/pSA+changes

Just the location of the page and the page name need changing IMO but will approve the PR.

**Review by jasonmotha — APPROVED** [review]

**Review by sungeunbae — APPROVED** [review]

**Pull request merged**